flink-table-store-core/src/main/java/org/apache/flink/table/store/file/data/AppendOnlyCompactManager.java

```diff
@@ -58,10 +59,9 @@ public AppendOnlyCompactManager(
     }
 
     @Override
-    public void submitCompaction() {
+    public void triggerCompaction() {
         if (taskFuture != null) {
-            throw new IllegalStateException(
```

Comment:
We could log the warn that there is uncompleted previous compaction before submitting new one.

Reply:
No need to warn, this is a happy path.